### PR TITLE
feat(16306): Complete BLS tests

### LINKED ISSUE #16306: Complete BLS tests
- split test between functional and technical
- use RNG util
- ensure all tests in [doc](https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/docs/proposals/TSS-Library/TSS-Library-Test-Plan.md) are implemented

### PR DESCRIPTION
**Pull request by mxtartaglia-sl**
**Description**:
Complete bls tests

**Related issue(s)**:

Fixes #https://github.com/hashgraph/hedera-services/issues/16306

### CODE REVIEW

File: cryptography/hedera-cryptography-altbn128/src/testFixtures/java/com/hedera/cryptography/altbn128/test/fixtures/AltBn128ExternalData.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public List<BigInteger> getScalars() {
+        var sksValues = (List<String>) data.get("SCALARS");
+        return sksValues.stream().map(BigInteger::new).toList();
+    }
+
+    /**
+     * Returns a list of g1 points generated with an external library.
+     * @return a list of g1 points generated with an external library.
+     */
+    public List<List<BigInteger[]>> getG1Points() {
```
Comment (lpetrovic05) [inline cryptography/hedera-cryptography-altbn128/src/testFixtures/java/com/hedera/cryptography/altbn128/test/fixtures/AltBn128ExternalData.java]: Maybe its worth introducing some like:
`record Coordinates(List<BigInteger> x, List<BigInteger> y)`

File: cryptography/hedera-cryptography-bls/src/test/java/com/hedera/cryptography/bls/SignatureSchemaTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    void testFindAltBn128Provider() {
+        assertDoesNotThrow(() -> PairingFriendlyCurves.findInstance(KnownCurves.ALT_BN128));
+        assertEquals(
+                KnownCurves.ALT_BN128,
+                PairingFriendlyCurves.findInstance(Curve.ALT_BN128).curve(),
+                "The pairing friendly curve should be ALT_BN128");
+    }
+
+    @ParameterizedTest
+    @MethodSource("combinedParameters")
```
Comment (lpetrovic05) [inline cryptography/hedera-cryptography-bls/src/test/java/com/hedera/cryptography/bls/SignatureSchemaTest.java]: I think you can get rid of this method and just use `EnumSource`

File: cryptography/hedera-cryptography-pairings-api/src/testFixtures/java/com/hedera/cryptography/pairings/test/fixtures/Utils.java [hunk 3]
```diff
@@ -143,4 +143,24 @@ public class Utils {
...
         return elements;
     }
+
+    /**
+     * Calculates the positive exponentiation of an integer
+     * @param value the value to exponenciate
+     * @param exponent the exponent
+     * @return the potentiated value
+     */
+    public static long intPow(final int value, final long exponent) {
```
Comment (lpetrovic05) [inline cryptography/hedera-cryptography-pairings-api/src/testFixtures/java/com/hedera/cryptography/pairings/test/fixtures/Utils.java]: doesn't the class `Math` have something like this?

Comment (mxtartaglia-sl) [reply cryptography/hedera-cryptography-pairings-api/src/testFixtures/java/com/hedera/cryptography/pairings/test/fixtures/Utils.java]: Not with longs, supports doubles, and casting has some rounding issues in specific powers:
https://stackoverflow.com/a/29996873.
Either way there are more performant implementations.

File: cryptography/hedera-cryptography-bls/src/test/java/com/hedera/cryptography/bls/SignatureSchemaTest.java (hunk 2 above)
Comment (mxtartaglia-sl) [reply cryptography/hedera-cryptography-bls/src/test/java/com/hedera/cryptography/bls/SignatureSchemaTest.java]: Perfect, thanks!

File: cryptography/hedera-cryptography-altbn128/src/testFixtures/java/com/hedera/cryptography/altbn128/test/fixtures/AltBn128ExternalData.java (hunk 1 above)
Comment (mxtartaglia-sl) [reply cryptography/hedera-cryptography-altbn128/src/testFixtures/java/com/hedera/cryptography/altbn128/test/fixtures/AltBn128ExternalData.java]: Nice, yes

**Review by rbarker-dev — APPROVED** [review]
LGTM

**Review by andrewb1269 — APPROVED** [review]
Review applies to the following files:
`.gitignore`
`cryptography/hedera-cryptography-altbn128/build.gradle.kts`
`cryptography/hedera-cryptography-bls/build.gradle.kts`
`hedera-dependency-versions/build.gradle.kts`

**Review by lpetrovic05 — APPROVED** [review]

**Pull request merged**